targets/disc-configs/lenovo-x1-carbon-disk-config.nix

```diff
@@ -0,0 +1,166 @@
+# Copyright 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+_: {
+  disko.devices = {
+    disk.disk1 = {
+      device = "DRIVE_PATH";
+      type = "disk";
+      content = {
+        type = "gpt";
+        partitions = {
+          boot = {
+            name = "boot";
+            size = "1M";
+            type = "EF02";
+          };
+          esp_a = {
+            name = "ESP_A";
+            size = "500M";
+            type = "EF00";
+            content = {
+              type = "filesystem";
+              format = "vfat";
+              mountpoint = "/boot";
+              mountOptions = ["umask=0077"];
+            };
+          };
+          esp_b = {
+            name = "ESP_B";
+            size = "500M";
+            type = "EF00";
+            content = {
+              type = "filesystem";
+              format = "vfat";
+              mountOptions = ["umask=0077"];
+            };
+          };
+          # LVM pool that is going to be encrypted
```

Comment:
How is the encryption done here? Would it be not cryptsetup -> lvm?

Reply:
There is some discussion going about that in the mail thread. I do not know at this moment how it is going to be implemented, and I decided to remove this comment to avoid the confusion.